src/screens/login/components/functions/oidc.ts

```diff
@@ -0,0 +1,177 @@
+import Minilog from '@cozy/minilog'
+import { Linking } from 'react-native'
+import type { WebViewNavigation } from 'react-native-webview'
+
+import { getErrorMessage } from '/libs/functions/getErrorMessage'
+import { closeInAppBrowser, showInAppBrowser } from '/libs/intents/InAppBrowser'
+
+const log = Minilog('CozyWebView.functions')
+
+const OIDC_PARAM = 'oidc'
+const USER_CANCELED = 'USER_CANCELED'
+const INVALID_CALLBACK = 'INVALID_CALLBACK'
+
+const OIDC_CALLBACK_URL_PARAM = 'redirect_after_oidc'
+const OIDC_CALLBACK_URL = 'https://links.mycozy.cloud/flagship/oidc_result'
+
+const OIDC_ONBOARD_CALLBACK_URL_PARAM = 'redirect'
+const OIDC_ONBOARD_CALLBACK_URL =
+  'https://links.mycozy.cloud/flagship/oidc_onboarding_finished'
+
+interface OidcLoginCallback {
+  code: string
+  fqdn: string
+  defaultRedirection: string | null
+}
+
+interface OidcOnboardingStartCallback {
+  code: string
+  onboardUrl: string
+}
+
+type OidcCallback = OidcLoginCallback | OidcOnboardingStartCallback
+
+interface OidcOnboardingEndCallback {
+  fqdn: string
+  onboardedRedirection: string | null
+}
+
+/**
+ * Check if the given NavigationRequest should trigger the OIDC scenario
+ */
+export const isOidcNavigationRequest = (
+  request: WebViewNavigation
+): boolean => {
+  try {
+    const url = new URL(request.url)
+
+    const oidcParam = url.searchParams.get(OIDC_PARAM)
+    const isOIDC = oidcParam !== null && oidcParam !== 'false'
+
+    return isOIDC
+  } catch (error) {
+    const errorMessage = getErrorMessage(error)
+    log.error(
+      `Something went wrong while trying to extract OIDC info from URL: ${errorMessage}`
```

Comment:
you should also log the `url`

Reply:
Fixed in https://github.com/cozy/cozy-flagship-app/compare/c351d6a960edac9811994f000ba5a790dcd992bf..02727db31d8c486e4afd0241610cd83f87ebf22a